apps/typegpu-docs/src/content/docs/guides/tgpu-cli.mdx

```diff
@@ -0,0 +1,521 @@
+```
+
+This will generate the TypeGPU definitions in the `different/path` directory with the `output.ts` filename.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader.wgsl
+  - <FileDiff> different/path </FileDiff>
+      - <FileDiff> output.ts </FileDiff>
+</FileTree>
+</div>
+
+It also supports glob patterns for specifying the output path:
+
+```bash
+tgpu-cli gen "path/to/*.wgsl" --output "output/*.ts"
+```
+
+This will generate the TypeGPU definitions in the `output` directory with the same names as the shaders, but with the `.ts` extension.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader1.wgsl
+      - shader2.wgsl
+      - example
+          - shader3.wgsl
+  - <FileDiff> output </FileDiff>
+      - <FileDiff> shader1.ts </FileDiff>
+      - <FileDiff> shader2.ts </FileDiff>
+</FileTree>
+</div>
+
+:::caution
+If you use a glob pattern for the input path, specifically `**`, make sure to use the `**` pattern
+in the output path as well to preserve the directory structure (*or make sure no conflicts will happen*).
+If you don't, all generated files will be placed in the same directory which may lead to name conflicts.
+
+<FileTree>
+
+- src
+  - example
+    - shader1.wgsl
+  - shader1.wgsl
+  - shader2.wgsl
+
+</FileTree>
+
+If you run:
+
+```bash
+tgpu-cli gen "src/**/*.wgsl" --output "output/*.ts"
+```
+
+The script will fail because all generated files will be placed in the `output` directory, and there will
+be a name conflict as both `shader1.wgsl` and `example/shader1.wgsl` will try to generate `output/shader1.ts`.
+
+<div class="relative">
+<FileTree>
+  - src
+      - example
+          - shader1.wgsl
+      - shader1.wgsl
+      - shader2.wgsl
+  - <FileDiff> output </FileDiff>
+      - <FileDiff> shader1.ts </FileDiff> \<\-\- name conflict
+      - <FileDiff> shader1.ts </FileDiff> \<\-\- name conflict
+      - <FileDiff> shader2.ts </FileDiff>
+</FileTree>
+</div>
+:::
+
+You can also use the `-o` shorthand for the `--output` option:
+
+```bash
+tgpu-cli gen "path/to/shader.wgsl" -o "different/path/output.ts"
+```
+
+### Supported extensions and module formats
+
+By default, the generated TypeScript files use the `.ts` extension and the ES module format.
+You can specify a different extension by just providing the extension in the output path:
+
+```bash
+tgpu-cli gen "path/to/shader.wgsl" -o "different/path/output.js"
+```
+
+This will generate the TypeGPU definitions in the `different/path` directory with the `output.js` filename.
+The supported extensions are:
+
+import ListItem from '../../../components/ListItem.astro';
+
+<ListItem idx={'1'}>
+  .js
+</ListItem>
+<ListItem idx={2}>
+  .cjs
+</ListItem>
+<ListItem idx={3}>
+  .mjs
+</ListItem>
+<ListItem idx={4}>
+  .ts
+</ListItem>
+<ListItem idx={5}>
+  .cts
+</ListItem>
+<ListItem idx={6}>
+  .mts
+</ListItem>
+
+You can also specify the module format using the `--commonjs` option:
```

Comment:
please make it clearer, that this option is the only way to generate CommonJS style files, providing .cjs ot .cts extension is not enough